ods_ci/tests/Tests/100__deploy/130__operators/140__serverless_operator/140__serverless_operator.robot

```diff
@@ -0,0 +1,116 @@
+    Wait For Pods Numbers  2    namespace=${KNATIVESERVING_NS}
+    ...    label_selector=app.kubernetes.io/component=controller    timeout=120
+    ${pod_names}=    Get Pod Names    ${KNATIVESERVING_NS}    app.kubernetes.io/component=controller
+    Verify Containers Have Zero Restarts    ${pod_names}    ${KNATIVESERVING_NS}
+    ${podname}=    Get Pod Name   ${OPERATOR_NAMESPACE}    label_selector=name=rhods-operator
+    Check For Errors On Operator Logs    ${podname}    ${OPERATOR_NAMESPACE}
+    Check DSC Conditions    ${KNATIVESERVING_NS}    default-dsc
+
+
+*** Keywords ***
+Suite Setup
+    [Documentation]    Suite Setup
+    Set Library Search Order    SeleniumLibrary
+    RHOSi Setup
+
+Suite Teardown
+    [Documentation]    Suite Teardown
+    Close All Browsers
+    RHOSi Teardown
+
+Assign Vars According To Product
+    [Documentation]    Assign vars related to product
+    [Arguments]    ${product}
+    IF    "${product}" == "RHODS"
+        Set Suite Variable    ${OPERATOR_APPNAME}     Red Hat OpenShift AI
+        Set Suite Variable    ${OPERATOR_NAME}    Red Hat OpenShift AI
+        Set Suite Variable    ${OPERATOR_NAMESPACE}    redhat-ods-operator
+    ELSE IF    "${product}" == "ODH"
+        Set Suite Variable    ${OPERATOR_APPNAME}  Open Data Hub Operator
+        Set Suite Variable    ${OPERATOR_NAME}    Open Data Hub Operator
+        Set Suite Variable    ${OPERATOR_NAMESPACE}    openshift-operators
+    END
+
+Is Resource Present
+    [Documentation]    Check CR
+    [Arguments]       ${resource}     ${resource_name}    ${namespace}
+    ${rc}=     Run and Return Rc
+    ...  oc get ${resource} ${resource_name} -n ${namespace}
+    Should Be Equal    "${rc}"    "0"    msg=${resource} does not exist
+
+Check Status
+    [Documentation]    Check Resource Status
+    [Arguments]       ${oc_get}     ${resource}    ${expected_status}
+    ${status}=     Run
+    ...  ${oc_get}
+    Log    ${status}    console=True
+    Should Be Equal    ${status}    ${expected_status}   msg=${resource} is not in Ready status
+
+Check DSC Conditions
+    [Documentation]    Checks that all conditions Reconciled Succesfully
+    [Arguments]    ${namespace}    ${dsc_name}
+    ${rc}    ${out}=    Run And Return Rc And Output
+    ...    oc get DataScienceCluster ${dsc_name} -n ${namespace} -o jsonpath='{.status.conditions[].reason}'
+    Should Be Equal As Integers    ${rc}     ${0}
+    Log    ${out}    console=${out}
+
+Check For Errors On Operator Logs
+    [Documentation]    Checks there are no errors on Operator Logs
+    [Arguments]    ${operator_name}    ${operator_namespace}
+    ${pod_logs}=    Oc Get Pod Logs  name=${operator_name}  namespace=${operator_namespace}  container=rhods-operator
+    ${error_present}=    Run Keyword And Return Status    Should Contain    ${pod_logs}    ERROR
+    IF    ${error_present}
+        Log    message=Check Pod Logs, ERROR level logs found.    level=WARN
```

Comment:
Just warning here? Errors in operator logs are just fine?

Reply:
There are ERROR level logs at the start because of Reconciliation sync, then it is fine

Reply:
Yes, so shouldn't we limit to check logs after some time when the operator is stable? Otherwise, do you plan to open these test results even when they are green and search for the content of this?

Also, it would be nice if operator would be smart enough to try to understand the context and reduce emitting errors at the beginning when components are starting. But I'm new to the OpenShift world, so my assumption may be just wrong.

Reply:
@zdtsw This is something that maybe could be improved on the operator side
In the meanwhile, we need a test in place that is not prone to errors

Reply:
which part need to be done in operator?

Reply:
Not printing ERROR logs due to Reconcile if afterward everything goes fine
e.g.
```
2024-03-07T10:12:03Z	ERROR	Reconciler error	{"controller": "secret-generator-controller", "controllerGroup": "", "controllerKind": "Secret", "Secret": {"name":"dashboard-oauth-config","namespace":"redhat-ods-applications"}, "namespace": "redhat-ods-applications", "name": "dashboard-oauth-config", "reconcileID": "65761c63-8b59-44bc-8a57-4471f6163df0", "error": "Secret \"dashboard-oauth-config-generated\" not found"}
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/remote-source/operator/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:329
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/remote-source/operator/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/remote-source/operator/deps/gomod/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235
```
I see this error everytime Operator gets installed

Reply:
@asanzgom regarding this test - so no plan to wait until operator is stable and check only the portion of the logs after that?

Also, looks like it just prints this warning message but doesn't print any actual error in the log... it's kind of useless, isn't it? :thinking:

Reply:
We need to check the Operator's Pod Logs for errors. There is one Test Case that could be refactored into a reusable Keyword for this 
ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/135__rhods_operator_logs_verification.robot

Reply:
I agree with Jan's points.
`Check For Errors On Operator Logs` so I would expect it failing if any error is found. I understand that ERRORS at the startup are accepted/expected, but what is the benefit we got from having this check?

Reply:
Refactored to fail when ERROR reg ex is found on logs and added ProductBug Tag